client/state/initial-state.js

```diff
@@ -88,14 +87,17 @@ export function persistOnChange( reduxStore, serializeState = serialize ) {
 }
 
 export default function createReduxStoreFromPersistedInitialState( reduxStoreReady ) {
-	if ( config.isEnabled( 'persist-redux' ) && ! isSupportUserSession() ) {
-		localforage.getItem( 'redux-state' )
-			.then( loadInitialState )
-			.catch( loadInitialStateFailed )
-			.then( persistOnChange )
-			.then( reduxStoreReady );
-	} else {
-		debug( 'persist-redux is not enabled, building state from scratch' );
-		reduxStoreReady( loadInitialState( {} ) );
-	}
+	require.ensure( 'state', () => {
```

Comment:
On the topic of refactoring `asyncRequire`, I got to thinking last week why it wasn't implemented as a promise, in which case if you'd wanted to have asynchronously loaded `state/index` and `state/reducer` simultaneously, it could be:

```js
const [ createReduxStore, reducer ] = await Promise.all( [ 
	asyncRequire( 'state' ),
	asyncRequire( 'state/reducer' )
] );
```